openmc/plotter.py

```diff
@@ -540,3 +602,279 @@ def _calculate_xs_elem_mat(this, types, temperature=294., cross_sections=None,
+
+    if xsdata is not None:
+        # Obtain the nearest temperature
+        t = (np.abs(xsdata.temperatures - temperature)).argmin()
+
+        # Get the data
+        data = np.zeros((len(types), library.energy_groups.num_groups))
+        for i, line in enumerate(types):
+            if 'fission' in line and not xsdata.fissionable:
+                continue
+            elif line == 'unity':
+                data[i, :] = 1.
+            else:
+                temp_data = getattr(xsdata, _PLOT_MGXS_ATTR[line])[t]
+                shape = temp_data.shape[:]
+                # If we have angular data, then we will plot the average
+                # over all provided angles.  Since the angles are equi-distant,
+                # un-weighted averaging will suffice
+                if xsdata.representation == 'angle':
+                    temp_data = np.mean(temp_data, axis=(0, 1))
+                if shape in (xsdata.xs_shapes["[G']"],
+                             xsdata.xs_shapes["[G]"]):
+                    data[i, :] = temp_data
+                elif shape == xsdata.xs_shapes["[G][G']"]:
+                    data[i, :] = np.sum(temp_data, axis=1)
+                elif shape == xsdata.xs_shapes["[DG]"]:
+                    if orders[i]:
+                        if orders[i] < len(shape[0]):
+                            data[i, :] = temp_data[orders[i]]
+                    else:
+                        data[i, :] = np.sum(temp_data[:])
+                elif shape in (xsdata.xs_shapes["[G'][DG]"],
+                               xsdata.xs_shapes["[G][DG]"]):
+                    if orders[i]:
+                        if orders[i] < len(shape[1]):
+                            data[i, :] = temp_data[:, orders[i]]
+                    else:
+                        data[i, :] = np.sum(temp_data[:, :], axis=1)
+                elif shape == xsdata.xs_shapes["[G][G'][DG]"]:
+                    temp_data = np.sum(temp_data, axis=1)
+                    if orders[i]:
+                        if orders[i] < len(shape[1]):
+                            data[i, :] = temp_data[:, orders[i]]
+                    else:
+                        data[i, :] = np.sum(temp_data[:, :], axis=1)
+                elif shape == xsdata.xs_shapes["[G][G'][Order]"]:
+                    temp_data = np.sum(temp_data, axis=1)
+                    if orders[i]:
+                        order = orders[i]
+                    else:
+                        order = 0
+                    if order < shape[1]:
+                        data[i, :] = temp_data[:, order]
+    else:
+        raise ValueError("{} not present in provided MGXS "
+                         "library".format(this.name))
+
+    return data
+
+
+def _calculate_mgxs_elem_mat(this, types, library, orders=None,
+                             temperature=294., ce_cross_sections=None,
+                             enrichment=None):
+    """Determines the multi-group cross sections of an element or material
```

Comment:
I didnt know the sentence was strict. Either way, all docstrings in this module have it now